network/p2p/middleware.go

```diff
@@ -602,6 +595,54 @@ func (m *Middleware) Unsubscribe(channel channels.Channel) error {
 	return nil
 }
 
+// processUnicastStreamMessage will decode, perform authorized sender validation and process a message
+// sent via unicast stream. This func should be invoked in a separate goroutine to avoid creating a message decoding bottleneck.
+func (m *Middleware) processUnicastStreamMessage(ctx context.Context, remotePeer peer.ID, msg *message.Message) {
+	channel := channels.Channel(msg.ChannelID)
+
+	decodedMsgPayload, err := m.codec.Decode(msg.Payload)
+	// collect slashing information for messages sent with unknown an message code byte
+	if codec.IsErrUnknownMsgCode(err) {
+		// slash peer if message contains unknown message code byte
+		violation := &slashing.Violation{PeerID: remotePeer.String(), Channel: channel, IsUnicast: true, Err: err}
+		m.slashingViolationsConsumer.OnUnknownMsgTypeError(violation)
+		return
+	}
+
+	// unexpected error condition. this indicates there's a bug
+	// don't crash as a result of external inputs since that creates a DoS vector
+	if err != nil {
+		m.log.
+			Error().
+			Err(fmt.Errorf("unexpected error while decoding message: %w", err)).
+			Hex("sender", msg.OriginID).
+			Hex("event_id", msg.EventID).
+			Str("event_type", msg.Type).
+			Str("channel", msg.ChannelID)
```

Comment:
This needs a `Msg` or `Send` at the end or the event will never be sent

Reply:
https://github.com/onflow/flow-go/pull/2832/commits/d963ffd307949786e764824e5539fc81db9297f5